### PR TITLE
Add xdebug profiling support

### PR DESCRIPTION
**Pull request by mikeymike**
Adds support for profiling. 

Unable to test compile due to a bunch of I think unrelated errors. 

Also this relies on new [PHP image builds](https://github.com/WeareJH/docker-m2-recipe/pull/23)

### CODE REVIEW

**Comment by shakyShane** [comment]
@mikeymike do you mean errors in the PHP builds, because the builds here look good

**Comment by mikeymike** [comment]
Nah my local builds... I need to sort out my Rust installation again

**Comment by shakyShane** [comment]
@mikeymike so this is ready to go? not sure if anythings else is needed etc

**Comment by mikeymike** [comment]
I think it's ready, images have been released a while back now 😄

**Comment by shakyShane** [comment]
@mikeymike have you considered how this interacts with ‘wf2 xdebug enable’ etc? Or perhaps it has no effect?

**Comment by mikeymike** [comment]
That's a good point, I didn't really look at the implementation details around that now you mention it. 

This updates the NGINX though so maybe it does?

**Comment by mikeymike** [comment]
So I just had a look at the implementation of `xdebug enable/disable` and I they're compatible. 

This PR introduces a `map` to choose debug or non debug upstreams. 
The current xdebug enable command keeps the same upstream keys but changes where they go so they're both pointed to `php-debug`. 

So in the event of running `xdebug enable` regardless of where the map decides the upstream is we'll still always end up in the debug container anyway which is the intended result. 

tldr looks good to me
